### PR TITLE
Van 7907 crm implementation

### PR DESCRIPTION
**Pull request by stephengreenfield**
CRM functions for adding customer and card info

### CODE REVIEW

File: lib/direct_connect/direct_connect_gateway.rb [hunk 1]
```diff
@@ -179,18 +262,28 @@ def parse(action, body)
         doc.remove_namespaces!
         response = {action: action}
 
+        service = actionToService(action)
+
         # special parsing
-        response[:result] = doc.at_xpath("//Response/Result").content.to_i
+        case service
+          when :manageCustomer
+            response[:result] = doc.at_xpath("//RecurringResult/code").content.to_s == 'OK' ? 0 : nil
```
Comment (BaconSoap) [inline lib/direct_connect/direct_connect_gateway.rb]: at some point, we're going to need to pass through/map whatever custom errors there this returns

**Comment by BaconSoap** [comment]
overall :+1:. Does this hit also a different user story? If so, could you mention the JIRA issue?

**Comment by stephengreenfield** [comment]
@NGPVAN/yellow code review

**Comment by afrieden** [comment]
Looks pretty good. I think this covers parts of VAN-7907 and VAN-7908

File: lib/direct_connect/direct_connect_gateway.rb [hunk 2]
```diff
@@ -128,15 +129,97 @@ def scrub(transcript)
...
+        post[:dayphone] = options[:dayphone]
+        post[:nightphone] = options[:nightphone]
+        post[:fax] = options[:fax]
+        post[:mobile] = options[:mobile]
+        post[:status] = options[:status]
+        post[:extdata] = nil
+      end
 
+      def add_credit_card_info(post, options)
+        add_authentication(post, options)
```
Comment (afrieden) [inline lib/direct_connect/direct_connect_gateway.rb]: I think this should get moved to an explicit step in add_card and update_card to follow how authentication gets added in other functions that hit direct connect (like in delete_customer)

**Comment by jackdolabany** [comment]
Like i was saying in hipchat, lets see if we can find time to convert these options hashes to classes this sprint

**Comment by jackdolabany** [comment]
Looks good :womans_hat:

**Comment by BaconSoap** [comment]
once you are happy with this, feel free to merge the pull request

**Pull request merged**